pkg/elle/rw_register/op.go

```diff
@@ -0,0 +1,77 @@
+package rwregister
+
+import (
+	"regexp"
+	"strconv"
+
+	"github.com/pingcap/tipocket/pkg/elle/core"
+)
+
+var (
+	opPattern = regexp.MustCompile(`([rw])([a-zA-Z])([0-9_]+)(.*)`)
+)
+
+// IntPtr copy int and return its pointer
+func IntPtr(i int) *int {
+	return &i
+}
+
+// MustParseOp ...
+func MustParseOp(opStr string) core.Op {
+	op := core.Op{
+		Type:  core.OpTypeOk,
+		Value: new([]core.Mop),
+	}
+
+	for opStr != "" {
+		opMatch := opPattern.FindStringSubmatch(opStr)
+		if len(opMatch) != 5 {
+			break
+		}
+		opStr = opMatch[4]
+		var (
+			mopType core.MopType
+			mopKey  = opMatch[2]
+		)
+		switch opMatch[1] {
+		case "r":
+			mopType = core.MopTypeRead
+		case "w":
+			mopType = core.MopTypeWrite
+		default:
+			panic("unreachable")
+		}
+		var mopVal *int
+		if opMatch[3] != "_" {
+			mopValInt, err := strconv.Atoi(opMatch[3])
+			if err != nil {
+				panic(err)
+			}
+			mopVal = &mopValInt
+		} else {
+			mopVal = IntPtr(initMagicNumber)
+		}
+		*op.Value = append(*op.Value, core.Mop{
+			T: mopType,
+			M: map[string]interface{}{
+				mopKey: mopVal,
+			},
+		})
+	}
+
+	return op
+}
+
+// Pair ...
+func Pair(op core.Op) (core.Op, core.Op) {
```

Comment:
How will this used? for testing?

Reply:
Yes, the functions in op will used by testing code only now.